ext/pdo_pgsql/pgsql_statement.c

```diff
@@ -386,7 +399,16 @@ static int pgsql_stmt_param_hook(pdo_stmt_t *stmt, struct pdo_bound_param_data *
 						S->param_formats[param->paramno] = 0;
 					}
 
-					if (PDO_PARAM_TYPE(param->param_type) == PDO_PARAM_LOB) {
+					if (PDO_PARAM_TYPE(param->param_type) == PDO_PARAM_INT) {
+						/* we need to check if the number requires bigints */
+						long long val = strtoll(Z_STRVAL_P(parameter), NULL, 10);
```

Comment:
Without having looked into Taylor's issue yet, this code is clearly broken :( If the original value was null or bool, then it will stay as such, while this code assumes the value is always a string. Passing a null to a PARAM_INT binding causes a segfault now (bool doesn't for some reason).

Reply:
Indeed I thought I had better coverage :(